packages/gestalt/src/SideNavigation/NavigationContent.js

```diff
@@ -88,62 +91,65 @@ export default function NavigationContent({
   const normalWidth = 280;
   const headerWidth = isCollapsed ? 44 : undefined;
   const collapsedWidth = shouldCollapseEmpty ? 40 : 60;
-
-  const wrapperWidth = sideNavigationCollapsed ? collapsedWidth : normalWidth;
   const contentWidth = isCollapsed ? collapsedWidth : normalWidth;
 
   return (
     <Box
+      ref={mainContainer}
       aria-label={accessibilityLabel}
       as="nav"
       color="default"
       height="100%"
       minWidth={collapsible ? undefined : normalWidth}
       position="relative"
-      width={collapsible ? wrapperWidth : undefined}
+      width={sideNavigationCollapsed ? collapsedContainerWidth : undefined}
       zIndex={overlayPreview ? new FixedZIndex(1) : undefined}
     >
       <div
         ref={scrollContainer}
-        className={classnames(styles.fullHeight, layoutStyles.borderBox, {
+        className={classnames(styles.fullHeight, layoutStyles.overflowAutoY, {
           [borderStyles.borderRight]: showBorder && !overlayPreview,
           [borderStyles.raisedBottom]: overlayPreview,
           [styles.contentWidthTransition]: collapsible,
-          [layoutStyles.overflowScrollY]: collapsible,
           [boxStyles.default]: collapsible,
         })}
-        style={{ width: collapsible ? contentWidth : undefined }}
+        style={{ width: collapsible ? 'max-content' : undefined }}
       >
-        {collapsible && <Collapser raised={isScrolled} />}
-
-        <Box
-          dangerouslySetInlineStyle={{
-            __style: {
-              paddingBottom: 24,
-            },
-          }}
-          display={shouldHideItems ? 'none' : undefined}
-          padding={2}
-          width={collapsible ? contentWidth : undefined}
+        <div
+          className={classnames({ [styles.contentWidthTransition]: collapsible })}
+          style={{ width: collapsible ? contentWidth : undefined }}
```

Comment:
I had to introduce 3rd inner wrapper. Its purpose is to preserve the content with in a given state (collapsed: 60px, expanded: 280px). It also is responsible for the transition animation.